### DISCUSSION TITLE
How to change which material is pointed to by Handle<ColorMaterial>

### DISCUSSION THREAD

**Discussion by ViddeM** [bevyengine/bevy#12383] [Q&A] [answered]
Hi! 

I'm trying to build a game where I have a bunch (order of millions) of tiles that can have a relatively small number of colors (~10). The tiles are all spawned using `MaterialMesh2dBundle` with the material set to an initial choice. However, throughout the game the tiles are supposed to be able to change to one of the other materials and I can't seem to figure out how to do this.

Of course I could have a unique material for each tile but that seems highly inefficient when what I really would like to do is simply point the existing material handle to a different material handle.

**Comment by alice-i-cecile** [comment, marked as answer]
I would store a central resource that lets you quickly and easily look up the handles:

```rust
#[derive(PartialEq, Eq, Hash)]
enum TileType {
   Grass,
   Stone,
   Dirt,
}

#[derive(Resource, Deref, DerefMut)
struct TileHandles(HashMap<TileType, Handle<ColorMaterial>>);

// Set your handles like
*tile_handle = tile_handles.get(TileType::Grass).unwrap().clone_weak();
```

**Reply by ViddeM** [reply]
Ah yeah I already had the central storage setup, I just didn't realize I could just dereference the tile_handle to update it 😅

Thanks! :)
